components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/TenantAdminEmailVerificationHandler.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.identity.recovery.handler;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.common.model.User;
+import org.wso2.carbon.identity.base.IdentityRuntimeException;
+import org.wso2.carbon.identity.core.bean.context.MessageContext;
+import org.wso2.carbon.identity.core.handler.InitConfig;
+import org.wso2.carbon.identity.event.IdentityEventConstants;
+import org.wso2.carbon.identity.event.IdentityEventException;
+import org.wso2.carbon.identity.event.event.Event;
+import org.wso2.carbon.identity.recovery.IdentityRecoveryConstants;
+import org.wso2.carbon.identity.recovery.IdentityRecoveryException;
+import org.wso2.carbon.identity.recovery.RecoveryScenarios;
+import org.wso2.carbon.identity.recovery.RecoverySteps;
+import org.wso2.carbon.identity.recovery.util.Utils;
+import org.wso2.carbon.registry.core.utils.UUIDGenerator;
+import org.wso2.carbon.user.core.UserStoreManager;
+
+import java.util.Map;
+
+/**
+ * This handler is used to send password reset link for tenant tenant admin.
+ */
+public class TenantAdminEmailVerificationHandler extends UserEmailVerificationHandler {
+
+    private static final Log log = LogFactory.getLog(TenantAdminEmailVerificationHandler.class);
+
+    public String getName() {
+
+        return "tenantAdminEmailVerification";
+    }
+
+    public String getFriendlyName() {
+
+        return "Tenant admin Email Verification";
+    }
+
+    @Override
+    public void handleEvent(Event event) throws IdentityEventException {
+
+        String eventName = event.getEventName();
+
+        if (log.isDebugEnabled()) {
+            log.debug("Handling event : " + eventName);
+        }
+
+        Map<String, Object> eventProperties = event.getEventProperties();
+        UserStoreManager userStoreManager = (UserStoreManager) eventProperties.get(IdentityEventConstants
+                .EventProperty.USER_STORE_MANAGER);
+
+        if (IdentityEventConstants.Event.PRE_SET_USER_CLAIMS.equals(eventName)) {
```

Comment:
call handle method if and only if the event is PRE_SET_USER_CLAIMS and TENANT_ADMIN_ASK_PASSWORD_CLAIM is there in the request.